feat: [#765] Add pid handling for: HANDLE, PMID and PMC

wujuu opened the pull request:
Closes #765.

Also added a hotfix for doctype being visible again in publications.

Michal-Kolomanski commented on ui/apps/ui/src/app/collections/data/utils.ts:
```diff
@@ -70,21 +71,65 @@ export const formatPublicationDate = (
+const resolveHandleLink = (rawHandle: string): string | undefined => {
+  const handleRegex = new RegExp('^\\S+/\\S+$');
+  return handleRegex.test(rawHandle)
+    ? `https://hdl.handle.net/${rawHandle}`
```
Current PID:
- DOI: `https://doi.org/${rawDoi}`

This PR added:
- HANDLE: `https://hdl.handle.net/${rawHandle}`
- PMC: `https://europepmc.org/article/PMC/${urlPart}`
- PMID: `https://pubmed.ncbi.nlm.nih.gov/${rawPmid}`

Total statistics for PIDs:
```
has_doi     3947182
no pid      1093032
has_handle   169408
has_pmid      53261
has_pmc       23839
has_arXiv     12011
has_w3id       3051
has_pdb           2
```
- Shouldnt we also add arXiv and w3id PID? @agpul @wujuu 

@Marcelinna @agpul 
The PIDs that have been added, need to be tested on this PR. Furthermore, it is necessary to confirm that those links are proper for each PID. Maybe Anna Wałek could confirm this?

wujuu replied:
Based on @wiktorflorian 's research - those as the correct linka for these pids. We basically haven't found any counter examples where they wouldn't work

Michal-Kolomanski reviewed (approved):
Good job! @Marcelinna please check if those new added PIDs are visible and working

Marcelinna commented:
I think we don't want to merge this if the HANDLE identifier is so extensive @agpul @aleksandrahojszyk @wujuu 
![image](https://github.com/cyfronet-fid/eosc-search-service/assets/59872096/c97ba3ce-ed1a-469d-9291-6bbb556495da)

Marcelinna commented:
@agpul @wujuu I don't see PMC in ticket
![image](https://github.com/cyfronet-fid/eosc-search-service/assets/59872096/99bd6ce1-febf-436d-8b76-75e71294c4b8)

agpul commented:
> @wujuu Can you provide me a record with a PMD identifier? ![image](https://user-images.githubusercontent.com/59872096/267386937-e6d115fd-e74b-4204-ba00-3b4f1465e575.png)

Hmm I think it is a misspelling PMC, not PMD, sorry.

agpul commented:
> @wujuu Can you provide me a record with a PMD identifier? ![image](https://user-images.githubusercontent.com/59872096/267386937-e6d115fd-e74b-4204-ba00-3b4f1465e575.png)

same here PMC not PMD

agpul commented:
> I think we don't want to merge this if the HANDLE identifier is so extensive @agpul @aleksandrahojszyk @wujuu ![image](https://user-images.githubusercontent.com/59872096/267385253-c97ba3ce-ed1a-469d-9291-6bbb556495da.png) https://search-4.docker-fid.grid.cyf-kr.edu.pl/search/dataset?q=*&standard=false

@aleksandrahojszyk this need to be fixed somehow

agpul commented:
> > I think we don't want to merge this if the HANDLE identifier is so extensive @agpul @aleksandrahojszyk @wujuu ![image](https://user-images.githubusercontent.com/59872096/267385253-c97ba3ce-ed1a-469d-9291-6bbb556495da.png) https://search-4.docker-fid.grid.cyf-kr.edu.pl/search/dataset?q=*&standard=false
> 
> @aleksandrahojszyk this need to be fixed somehow

we agreed to add a show more button if there are more than 4 identifiers.

aleksandrahojszyk commented:
<img width="833" alt="Screenshot 2023-09-15 at 10 19 15" src="https://github.com/cyfronet-fid/eosc-search-service/assets/49695272/ee4cbda1-6460-4b1a-aad9-130861024608">

wujuu commented:
@Marcelinna @aleksandrahojszyk I've added "show more" feature to all tags - but it's only enabled in identifiers for now.

Also changed the regex for HANDLE because some of the datasets had straight-up url instead of an identifier. We can click on that as well now.

Marcelinna commented:
> @Marcelinna @aleksandrahojszyk I've added "show more" feature to all tags - but it's only enabled in identifiers for now.
> 
> Also changed the regex and url construction for HANDLE because some of the datasets had straight-up url instead of an identifier. We can click on that as well now.

@aleksandrahojszyk we want add this to all tags ?
![image](https://github.com/cyfronet-fid/eosc-search-service/assets/59872096/8e47ace4-9d5f-4174-b088-cda878d21b88)

Marcelinna commented:
@wujuu  the same color?
![image](https://github.com/cyfronet-fid/eosc-search-service/assets/59872096/c6821688-a3ec-43ff-ba09-ee8b2f554697)

aleksandrahojszyk commented:
SORRY I made a mistake
when folded for example -> + 15 
when unfolded -> Show less 
color default 30549F 
yes, with authors and keywords the same mechanics
3 lines of keywords and 3 lines of authors

wujuu commented:
@Marcelinna @aleksandrahojszyk  Ok, I'll add the show more func for authors as well - for keywords (they are secondary tags) it's not implemented yet

wujuu commented:
Added "show more" feat to authors. Changed from "+ X more" to "+ X". Changed the font weight on the button, because the color was the same as specified by @aleksandrahojszyk, although to me it still looks off
